modules/chemical_reactions/include/utils/ThermochimicaUtils.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+class InputParameters;
+class MooseObject;
+class Action;
+
+#include <string>
+
+namespace ThermochimicaUtils
```

Comment:
should this be a framework-wide thing to handle external contribs that may or may not be compiled in

Reply:
Hm, I don't see how this could be generalized yet. And a central file in framework for optional contribs that are on modules seems more confusing than helpful to me.

Reply:
OptionalContribInterface base class for all objects tied to an external contrib
keeps these 3 routines. They would need to be re-implemented a bit to not rely on pre-processing. 

but you could keep track of the optional contrib name in the interface too

Reply:
"Not to rely on preprocessing"... gee, that would make things a lot more complicated, and we would have to rely on preprocessing anyways in the actual objects that depend on the optional contribs. Please keep in mind that the primary goal of this PR is to unbreak Falcon so that documentation can be merged in _before the holiday curtailment_.

Reply:
> Please keep in mind that the primary goal of this PR is to unbreak Falcon so that documentation can be merged in before the holiday curtailment.

If that’s the goal… we just checkout the submodule…

Reply:
So what changes are requested?

Reply:
I m suggesting ideas if we dont want to have to replicate everything for Saline

Reply:
not really requesting anything right now, more like discussing

Reply:
> If that’s the goal… we just checkout the submodule…

I'd say this PR is superior to that approach as it fixes doc building for local users, too.

Reply:
Anyways after some thinking I just don't see a way to do this without the preprocessor.